### DISCUSSION TITLE
Sending multipart responses

### DISCUSSION THREAD

**Discussion by dubadub** [rwf2/Rocket#2809] [Questions] [answered]
Hi there,

I'm wondering how can I send multipart responses with Rocket? I would imagine I should be using Response builder, but I can't figure out how... Ideally with streaming...

Response headers:
```
Content-Type: multipart/x-mixed-replace; boundary=----MultipartStreamBoundary
```

Body:
```
----MultipartStreamBoundary
Content-Type: text/plain

How can I help you
----MultipartStreamBoundary
Content-Type: text/plain

today?
----MultipartStreamBoundary
Content-Type: audio/mpeg

���� BINARY DATA ����
```

Thank you!

**Comment by the10thWiz** [comment]
This turns out to be a bit more complex than most `Responder` implementations, since we are adapting multiple (different) readers into a single reader. I've started working on a proper solution, and I am planning to publish it as a crate.

**Reply by the10thWiz** [reply, marked as answer]
I've published my solution as [rocket-multipart](https://crates.io/crates/rocket-multipart), please let me know if you find any issues, or require additional features.

**Reply by dubadub** [reply]
Wow! That's really impressive! Thank you for you help. I'll give it a try in a few days.

**Reply by dubadub** [reply]
Hi @the10thWiz [it worked](https://github.com/cooklang/cooklang-sync/blob/main/server/src/chunks/mod.rs#L140-L156) really well for me, thank you! I'm wondering if it's easy enough to add custom headers into `MultipartSection`? I can try making a PR if you don't mind...

**Reply by the10thWiz** [reply]
Feel free to make a PR. I'll take a look tonight, but it should be quite easy to add custom headers.
